handler/src/main/java/io/netty/handler/ssl/OpenSsl.java

```diff
@@ -57,7 +60,22 @@
         // If in the classpath, try to load the native library and initialize netty-tcnative.
         if (cause == null) {
             try {
-                NativeLibraryLoader.load("netty-tcnative", SSL.class.getClassLoader());
+                String os = normalizeOs(SystemPropertyUtil.get("os.name", ""));
+                String arch = normalizeArch(SystemPropertyUtil.get("os.arch", ""));
+                String libName = "netty-tcnative-" + os + '-' + arch;
+                try {
+                    // First, try loading the platform-specific library. Platform-specific
+                    // libraries will be available if using a tcnative uber jar.
+                    NativeLibraryLoader.load(libName, SSL.class.getClassLoader());
```

Comment:
I am not a fan of statically linked libraries, if a critical vulnerability in ssl is disclosed it would require a new build of netty-tcnative, new package of our application and then installing that to our environments where as dynamically linked would only require an application restart after updating the system dependencies (the latter could be done by our 24/7 standby team if the disclosure happens on the weekend whereas the prior would require calling in a developer).